tests/test_gh.py

```diff
@@ -69,131 +68,163 @@ def test_get_issue_content_bad(mocker):
-            f.write(repos_json_content)
+    if input_type == "url":
 
-    clone_repo_mocker = mocker.patch(
-        "subprocess.run",
-        side_effect=mock_clone_repo,
-    )
-    actual_group_dict, actual_repos_dict = _get_broadcast_repos_dict(
-        url_to_repo_info=url_to_repo_info
+        def mock_clone_repo(cwd, **kwargs):
+            # already a temporary directory created by
+            #   _get_broadcast_repos_dict
+            source_dir = user_filesystem / "repo_info_dir_json"
+            group_json_content = (source_dir / "groups.json").read_text()
+            repos_json_content = (source_dir / "repos.json").read_text()
+            target_dir = cwd[-1]
+            with open(f"{target_dir}/groups.json", "w") as f:
+                f.write(group_json_content)
+            with open(f"{target_dir}/repos.json", "w") as f:
+                f.write(repos_json_content)
+
+        mock_clone_repo = mocker.patch(
+            "subprocess.run",
+            side_effect=mock_clone_repo,
+        )
+    if mock_cwd:
+        available_mockers["mocker_cwd"] = mocker.patch(
+            "pathlib.Path.cwd",
+            return_value=user_filesystem / mock_cwd,
+        )
+    if mock_home:
+        available_mockers["mocker_home"] = mocker.patch(
+            "pathlib.Path.home",
+            return_value=user_filesystem / mock_home,
+        )
+    default_expected_group_dict, default_expected_repos_dict = (
+        {
+            "odd_group": ["repo1", "repo3"],
+            "even_group": ["repo2", "repo4"],
+        },
+        {
+            "repo1": "https://github.com/user/repo1",
+            "repo2": "https://github.com/user/repo2",
+            "repo3": "https://github.com/user/repo3",
+            "repo4": "https://github.com/user/repo4",
+        },
     )
-    assert actual_group_dict == expected_group_dict
-    assert actual_repos_dict == expected_repos_dict
-    clone_repo_mocker.assert_called()
-    clone_repo_mocker.stop()
-    # C2: a directory input and JSON files exist in the directory.
-    #   Expect the broadcast repos dict is returned.
-    actual_group_dict, actual_repos_dict = _get_broadcast_repos_dict(
-        url_to_repo_info=user_filesystem / "repo_info_dir_json"
+    another_expected_group_dict, another_expected_repos_dict = (
+        {
+            "small_group": ["repo1", "repo2"],
+            "large_group": ["repo101", "repo102"],
+        },
+        {
+            "repo1": "https://github.com/user/repo1",
+            "repo2": "https://github.com/user/repo2",
+            "repo101": "https://github.com/user/repo101",
+            "repo102": "https://github.com/user/repo102",
+        },
     )
-    assert actual_group_dict == expected_group_dict
-    assert actual_repos_dict == expected_repos_dict
-    # C3: a directory input and YAML files exist in the directory.
-    #   Expect the broadcast repos dict is returned.
+    if expected_output == "default":
```

Comment:
this is ugly.  Can you just put the expected output explicitly in the paramatrize and get rid of this ugly logic?

We like to use lists in there for clarity so it looks like
```
[
( # case 1
[input1, input2], [expected1, expected2, expected3]),
( # case 2
[input1, input2], [expected1, expected2, expected3])
]
```